### DISCUSSION TITLE
Generate types for Swagger

### DISCUSSION THREAD

**Discussion by kodermax** [prisma/prisma#4651] [Q&A]
Can Prisma generate types for swagger with swagger attribute for each field?

**Comment by KevinDennyII** [comment]
> Can Prisma generate types for swagger with swagger attribute for each field?

We are experiencing the same issue.

Prisma generated the following for us:
```
export type PersonSocialMediaUncheckedCreateInput = {
    person_social_media_id?: string
    person_id: string
    social_media_name?: string | null
    social_media_type_id: string
    is_active?: boolean
    data_source_id: string
    created_date?: Date | string
    updated_date?: Date | string
    created_by_user_id: string
    updated_by_user_id: string
    org_id: string
  }
```

Our controller looks like this:
```
@Post(':personId/social-media-prisma')
  @ApiOperation({ summary: 'Adds a new social media record' })
  createSocialMedia(
    @Request() req,
    @Param('personId') personId: string,
    @Body() createPersonSocialMediaDto: Prisma.PersonSocialMediaUncheckedCreateInput
  ) {
    console.log(createPersonSocialMediaDto);
    return this.personSocialMediaService.createSocialMedia(personId, createPersonSocialMediaDto, req.user.authDetails);
  }
```

We are using NestJS.  In reading that the generated files should replace the additional DTO files you would created with TypeORM, Swagger does not seems to recognize the Prisma generated type as it would the additional DTO files.

Definitely an issues that needs to be looked at here
